synapse/lib/urlhelp.py

```diff
@@ -69,6 +69,48 @@ def chopurl(url):
     ret['path'] = pathrem
     return ret
 
+def zipurl(info):
```

Comment:
If we could avoid needing this, it would be nice. Perhaps if we move to annotating insecure mode via a scheme modification, we can just use conditional startswith() and replace() vs surgery?

Reply:
Wanted to positively log the path being restored; while dropping the parameters (which often could contain secret material). Easy to just chop it with a `?` and be done.